### PR TITLE
[Merged by Bors] - feat: port Topology.Algebra.FilterBasis

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by int-y1** [comment]
todo:

* [ ] `attribute [to_additive]` errors
* [ ] `ModuleFilterBasis.continuousSMul` type mismatch at `B.smul`
* [ ] `ModuleFilterBasis.ofBases` failed to synthesize instance `TopologicalSpace R`

File: Mathlib/Topology/Algebra/FilterBasis.lean [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+  add' : ∀ {U}, U ∈ sets → ∃ V ∈ sets, V + V ⊆ U
+  neg' : ∀ {U}, U ∈ sets → ∃ V ∈ sets, V ⊆ (fun x ↦ -x) ⁻¹' U
+  conj' : ∀ x₀, ∀ {U}, U ∈ sets → ∃ V ∈ sets, V ⊆ (fun x ↦ x₀ + x + -x₀) ⁻¹' U
+#align add_group_filter_basis AddGroupFilterBasis
+
+attribute [to_additive existing] GroupFilterBasis GroupFilterBasis.conj'
+  GroupFilterBasis.toFilterBasis
+
+/-- `GroupFilterBasis` constructor in the commutative group case. -/
+@[to_additive "`add_group_filter_basis` constructor in the additive commutative group case."]
```
Comment (dupuisf) [inline Mathlib/Topology/Algebra/FilterBasis.lean:75]:
```suggestion
@[to_additive "`AddGroupFilterBasis` constructor in the additive commutative group case."]
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Pull request closed without merging**